test/language/statements/class/fields-asi-5.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2017 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: ASI test in field declarations -- error when PropertyName "in" interpreted as index
+esid: sec-automatic-semicolon-insertion
+features: [class-fields]
+---*/
+
+class C {
+  x
+  in
+  y
+}
+
+assert.throws(ReferenceError, function() {
```

Comment:
this is probably not an error, the construct should succeed. 

You might need to assert the new instance has own properties x, in, and y.

As we discussed in person, this can expand to the other case evaluating the right side expression:

```js
var x = 0;
var y = [42];
var z = 1;
class C {
  a = x
  in
  y
  b = z
  in
  y
}
```